### ISSUE TITLE
Scheduler does not update tasks on backend

### ISSUE THREAD

**Issue by Minus20Five** [se701g2/Doto#261]
### Bug Summary: 
From JSDoc in [`TaskScheduler`](https://github.com/se701g2/Doto/blob/master/doto-frontend/src/components/pages/Calendar/TaskScheduler.js): 
`"The existing tasks may have their schedules modified"`

However, in the [`addTask`](https://github.com/se701g2/Doto/blob/eaf48f34ecacf15db011dae214d9b523038478dc/doto-frontend/src/components/pages/Calendar/Calendar.js#L70) method it doesn't seem that displaced tasks are updated on the server.

--- 

### Steps To Reproduce
1. Add a new task
2. Add another task with an earlier deadline that should occur before the first task

### Expected behaviour
First task startdate, enddate, reminderdate details updated on server.

### Observed Behaviour
First take not updated on server.

### Environment

- Version: All
- Operating System: macOS
- Browser (if any): Chrome

---

**Comment by Minus20Five** [comment]
Blocked by #274 274
